src/services/redisSessionStorage.ts

```diff
@@ -313,17 +322,155 @@ export default class RedisSessionStorage extends RedisStorageUtils
       .run();
   }
 
+  /**
+   * Insert a user in the list of blocked account
+   *
+   * @param fiscalCode id of the user
+   *
+   * @returns a promise with either an error or true
+   */
+  public setBlockedUser(fiscalCode: FiscalCode): Promise<Either<Error, true>> {
+    return new Promise<Either<Error, true>>(resolve => {
+      log.info(`Adding ${fiscalCode} to ${blockedUserSetKey} set`);
+      this.redisClient.sadd(blockedUserSetKey, fiscalCode, err =>
+        resolve(err ? left(err) : right(true))
+      );
+    });
+  }
+
+  /**
+   * Remove a user from the list of blocked account
+   *
+   * @param fiscalCode id of the user
+   *
+   * @returns a promise with either an error or true
+   */
+  public unsetBlockedUser(
+    fiscalCode: FiscalCode
+  ): Promise<Either<Error, boolean>> {
+    return new Promise<Either<Error, boolean>>(resolve => {
+      log.info(`Removing ${fiscalCode} from ${blockedUserSetKey} set`);
+      this.redisClient.srem(blockedUserSetKey, fiscalCode, (err, response) =>
+        resolve(
+          this.falsyResponseToError(
+            this.integerReply(err, response, 1),
+            new Error(
+              "Unexpected response from redis client deleting blockedUserKey"
+            )
+          )
+        )
+      );
+    });
+  }
+
+  /**
+   * Check if a user is blocked
+   *
+   * @param fiscalCode id of the user
+   *
+   * @returns a promise with either an error or a boolean indicating if the user is blocked
+   */
   public async isBlockedUser(
     fiscalCode: FiscalCode
   ): Promise<Either<Error, boolean>> {
-    return this.sismemberTask("BLOCKED-USERS", fiscalCode)
+    return this.sismemberTask(blockedUserSetKey, fiscalCode)
       .bimap(
         err => new Error(`Error accessing blocked users collection: ${err}`),
         result => result === 1
       )
       .run();
   }
 
+  /**
+   * Delete all user session data
+   * @param fiscalCode
+   */
+  public async delUserAllSessions(
+    fiscalCode: FiscalCode
+  ): Promise<Either<Error, boolean>> {
+    const sessionsOrError = await this.readSessionInfoKeys(fiscalCode);
```

Comment:
```suggestion
    const errorOrSessions = await this.readSessionInfoKeys(fiscalCode);
```

(error is the "left" part)